{
  "uid" : "4ef479bd53d9dc36",
  "name" : "登录反向用例测试",
  "fullName" : "cases.test_login.TestLogin#test_login_failed",
  "historyId" : "28fbee3f7849e5c692210a9f12e7fe96",
  "time" : {
    "start" : 1615360243962,
    "stop" : 1615360266208,
    "duration" : 22246
  },
  "status" : "failed",
  "statusMessage" : "AssertionError: assert False\n +  where False = <bound method BassPage.is_toast_show of <pages.page_action.PageAction object at 0x000001BF5004E668>>('登录验证失败，用户不存在！')\n +    where <bound method BassPage.is_toast_show of <pages.page_action.PageAction object at 0x000001BF5004E668>> = <pages.page_action.PageAction object at 0x000001BF5004E668>.is_toast_show",
  "statusTrace" : "self = <cases.test_login.TestLogin object at 0x000001BF5141E6D8>\nhandler = (<appium.webdriver.webdriver.WebDriver (session=\"1d18ff9a-4d59-454a-90c8-b95fe2a83172\")>, <pages.page_action.PageAction object at 0x000001BF5004E668>)\ndata = {'error_msg': '登录验证失败，用户不存在！', 'name': '登录-反向用例-用户名错误', 'password': '654321', 'username': '财友'}\n\n    @allure.story('钓鱼人APP自动化测试-登录模块-反向用例测试')\n    @allure.title('登录反向用例测试')\n    @pytest.mark.parametrize('data',failed_data)\n    def test_login_failed(self, handler, data):\n        logger.info('开始执行反向测试用例：{}'.format(data['name']))\n        handler[1].login(data['username'], data['password'])\n        # logger.info('选中用户协议框')\n        # handler[1].BassPage.checkbox_agree()\n        logger.info('登录操作完成')\n>       assert handler[1].is_toast_show(data['error_msg'])\nE       AssertionError: assert False\nE        +  where False = <bound method BassPage.is_toast_show of <pages.page_action.PageAction object at 0x000001BF5004E668>>('登录验证失败，用户不存在！')\nE        +    where <bound method BassPage.is_toast_show of <pages.page_action.PageAction object at 0x000001BF5004E668>> = <pages.page_action.PageAction object at 0x000001BF5004E668>.is_toast_show\n\ncases\\test_login.py:42: AssertionError",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ {
    "name" : "prepare_to_login",
    "time" : {
      "start" : 1615360229968,
      "stop" : 1615360238359,
      "duration" : 8391
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "hasContent" : false,
    "stepsCount" : 0,
    "shouldDisplayMessage" : false,
    "attachmentsCount" : 0
  }, {
    "name" : "handler",
    "time" : {
      "start" : 1615360184900,
      "stop" : 1615360229927,
      "duration" : 45027
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "hasContent" : false,
    "stepsCount" : 0,
    "shouldDisplayMessage" : false,
    "attachmentsCount" : 0
  } ],
  "testStage" : {
    "status" : "failed",
    "statusMessage" : "AssertionError: assert False\n +  where False = <bound method BassPage.is_toast_show of <pages.page_action.PageAction object at 0x000001BF5004E668>>('登录验证失败，用户不存在！')\n +    where <bound method BassPage.is_toast_show of <pages.page_action.PageAction object at 0x000001BF5004E668>> = <pages.page_action.PageAction object at 0x000001BF5004E668>.is_toast_show",
    "statusTrace" : "self = <cases.test_login.TestLogin object at 0x000001BF5141E6D8>\nhandler = (<appium.webdriver.webdriver.WebDriver (session=\"1d18ff9a-4d59-454a-90c8-b95fe2a83172\")>, <pages.page_action.PageAction object at 0x000001BF5004E668>)\ndata = {'error_msg': '登录验证失败，用户不存在！', 'name': '登录-反向用例-用户名错误', 'password': '654321', 'username': '财友'}\n\n    @allure.story('钓鱼人APP自动化测试-登录模块-反向用例测试')\n    @allure.title('登录反向用例测试')\n    @pytest.mark.parametrize('data',failed_data)\n    def test_login_failed(self, handler, data):\n        logger.info('开始执行反向测试用例：{}'.format(data['name']))\n        handler[1].login(data['username'], data['password'])\n        # logger.info('选中用户协议框')\n        # handler[1].BassPage.checkbox_agree()\n        logger.info('登录操作完成')\n>       assert handler[1].is_toast_show(data['error_msg'])\nE       AssertionError: assert False\nE        +  where False = <bound method BassPage.is_toast_show of <pages.page_action.PageAction object at 0x000001BF5004E668>>('登录验证失败，用户不存在！')\nE        +    where <bound method BassPage.is_toast_show of <pages.page_action.PageAction object at 0x000001BF5004E668>> = <pages.page_action.PageAction object at 0x000001BF5004E668>.is_toast_show\n\ncases\\test_login.py:42: AssertionError",
    "steps" : [ ],
    "attachments" : [ {
      "uid" : "5ed0cc112ac12d06",
      "name" : "log",
      "source" : "5ed0cc112ac12d06.txt",
      "type" : "text/plain",
      "size" : 22977
    } ],
    "parameters" : [ ],
    "hasContent" : true,
    "stepsCount" : 0,
    "shouldDisplayMessage" : true,
    "attachmentsCount" : 1
  },
  "afterStages" : [ {
    "name" : "handler::0",
    "time" : {
      "start" : 1615360277323,
      "stop" : 1615360278687,
      "duration" : 1364
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "hasContent" : false,
    "stepsCount" : 0,
    "shouldDisplayMessage" : false,
    "attachmentsCount" : 0
  } ],
  "labels" : [ {
    "name" : "story",
    "value" : "钓鱼人APP自动化测试-登录模块-反向用例测试"
  }, {
    "name" : "feature",
    "value" : "钓鱼人APP自动化测试-登录模块"
  }, {
    "name" : "epic",
    "value" : "钓鱼人APP自动化测试"
  }, {
    "name" : "tag",
    "value" : "@pytest.mark.usefixtures('prepare_to_login')"
  }, {
    "name" : "parentSuite",
    "value" : "cases"
  }, {
    "name" : "suite",
    "value" : "test_login"
  }, {
    "name" : "subSuite",
    "value" : "TestLogin"
  }, {
    "name" : "host",
    "value" : "LAPTOP-LSM6I36P"
  }, {
    "name" : "thread",
    "value" : "15808-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "cases.test_login"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ {
    "name" : "data",
    "value" : "{'name': '登录-反向用例-用户名错误', 'username': '财友', 'password': '654321', 'error_msg': '登录验证失败，用户不存在！'}"
  } ],
  "links" : [ ],
  "hidden" : true,
  "retry" : true,
  "extra" : {
    "categories" : [ ],
    "tags" : [ "@pytest.mark.usefixtures('prepare_to_login')" ]
  },
  "source" : "4ef479bd53d9dc36.json",
  "parameterValues" : [ "{'name': '登录-反向用例-用户名错误', 'username': '财友', 'password': '654321', 'error_msg': '登录验证失败，用户不存在！'}" ]
}